src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,2389 @@
+		{
+			//minimum 0.00,maximum 100.00, units percent, Mode0 will not be considered when the outside air relative humidity is below the value in this field.
+			// note If this field is blank, the lower constraint on outside air relative humidity will be 0.00% (default 0.00), the upper constraint on outside air relative humidity will be 100.00%, (default 100.00)
+			Minimum_Outside_Air_Relative_Humidity = min;
+			Maximum_Outside_Air_Relative_Humidity = max;
+			return true;
+		}
+		bool CMode::InitializeReturnAirTemperatureConstraints(Real64 min, Real64 max)
+		{
+			//will not be considered when the return air temperature is below the value in this field.
+			//If this field is blank, there will be no lower constraint on return air temperature
+			Minimum_Return_Air_Temperature = min;
+			Maximum_Return_Air_Temperature = max;
+			return true;
+		}
+		bool CMode::InitializeReturnAirHumidityRatioConstraints(Real64 min, Real64 max)
+		{
+			//minimum 0.00 maximum 0.10, units kgWater / kgDryAir
+			//note Mode0 will not be considerd when outside air absolute humidity is below the value in this field.
+			//note If this field is blank, the lower constraint on outside air humidity ratio will be 0.00 kgWater / kgDryAir., default 0.00
+			//the upper constraint on outside air humidity ratio will be 0.10 kgWater / kgDryAir, default 0.10
+			Minimum_Return_Air_Humidity_Ratio = min;
+			Maximum_Return_Air_Humidity_Ratio = max;
+			return true;
+		}
+		bool CMode::InitializeReturnAirRelativeHumidityConstraints(Real64 min, Real64 max)
+		{
+			//minimum 0.00,maximum 100.00, units percent, Mode0 will not be considered when the outside air relative humidity is below the value in this field.
+			// note If this field is blank, the lower constraint on outside air relative humidity will be 0.00% (default 0.00), the upper constraint on outside air relative humidity will be 100.00%, (default 100.00)
+			Minimum_Return_Air_Relative_Humidity = min;
+			Maximum_Return_Air_Relative_Humidity = max;
+			return true;
+		}
+
+
+		bool CMode::InitializeOSAFConstraints(Real64 minOSAF, Real64 maxOSAF)
+		{
+			//minimum 0.00, maximum 1.00, Outside air fractions below this value will not be considered.
+			//If this field is blank, the lower constraint on outside air fraction will be 0.00,default 0.10
+			Min_OAF = minOSAF;
+			Max_OAF = maxOSAF;
+			return true;
+		}
+		bool CMode::InitializeMsaRatioConstraints(Real64 minMsa, Real64 maxMsa)
+		{
+			//minimum 0.00, maximum 1.00, Supply air mass flow rate ratios below this value will not be considered.
+			//Supply air mass flow rate ratio describes supply air mass flow rate as a fraction of mass flow rate associated with the value in field : "System Maximum Supply Air Flow Rate".
+			//If this field is blank, the lower constraint on outside air fraction will be 0.00,default 0.10
+			Min_Msa = minMsa;
+			Max_Msa = maxMsa;
+			return true;
+		}
+		bool CMode::ValidPointer(int curve_pointer) {
```

Comment:
What's the point of this function?  Zero is already falsy in C++.

Reply:
Well this related to the question later on about what to do if we don't have a valid curve pointer. There are cases where the curve might not be set or required. Like in standby mode for example. So in that case the curve pointerID will be -1 at that point, so we have a check. I'm sure there is a more elegant solution to this, but i'm not loosing sleep over it (except I am ;-).